In [3]:
import numpy as np
import pandas as pd
import scipy.stats as stats
import matplotlib.pyplot as plt
import statsmodels.api as sm
import statsmodels.stats.proportion as smprop

QUIZ

Question 3 of 6

In [4]:
1-stats.poisson.cdf(7,4.8)

0.11333382935932945

Question 6 of 6

In [5]:
print(1-stats.poisson.cdf(5,3))

0.08391794203130343


In [6]:
print(1-stats.norm.cdf(30, loc=36, scale=3))
print(1-stats.norm.cdf(-2))
print(stats.norm.cdf(2))

0.9772498680518208
0.9772498680518208
0.9772498680518208


2.1 Discrete random variable

a) Let X be a stochastic variable. When running the Python-command stats.binom.pmf(4,
10, 0.6) Python returns 0.1115. written as:

In [7]:
print(stats.binom.pmf(4, 10, 0.6))

0.11147673600000006


What distribution is applied and what does 0.1115 represent?
It is binomial distrubtion

The distribution applied is the binomial distribution with n = 10 observations and
p = 0.6 the probability for success. The value, 0.1115 output by Python is the value
of the probability density function (pdf) for x = 4, hence the probability of getting
exactly 4 successes in 10 draws with replacement with a success probability of 60%

b) Let X be the same stochastic variable as above. The following are results from Python:

In [8]:
print(stats.binom.cdf(4,10,0.6))
print(stats.binom.cdf(5,10,0.6))

0.16623861760000014
0.36689674240000025


Calculate the following probabilities: P(X ≤ 5), P(X < 5), P(X > 4) and
P(X = 5).

In [14]:
n = 10
p = 0.6

P_X_leq_5 = stats.binom.cdf(5,n,p)
P_X_lt_5 = stats.binom.cdf(4,n,p)
P_X_gt_4 = 1 - stats.binom.cdf(4,n,p)
P_X_eq_5 = stats.binom.cdf(5,n,p) - stats.binom.cdf(4,n,p)

print(f"P(X <= 5): {P_X_leq_5:.4f}")
print(f"P(X < 5): {P_X_lt_5:.4f}")
print(f"P(X > 4): {P_X_gt_4:.4f}")
print(f"P(X = 5): {P_X_eq_5:.4f}")

P(X <= 5): 0.3669
P(X < 5): 0.1662
P(X > 4): 0.8338
P(X = 5): 0.2007


c) Let X be a stochastic variable. From Python we get:


In [10]:
print(stats.poisson.pmf(k=4,mu=3))

0.16803135574154085


What distribution is applied and what does 0.16803 represent?

The Poisson distribution and the value is the probability of getting x = 4 events per
interval when the average events per interval λ = 3 (i.e. the mean)

In [11]:
(-2.71*1**(1/4)) / 1/4

-0.6775

In [12]:
(math.factorial(14) / math.factorial(3) * math.factorial(11)) / math.factorial(20) / math.factorial(3) * math.factorial(17)

NameError: name 'math' is not defined

Let X be the same stochastic variable as above. The following are results
from Python:

In [ ]:
print(stats.poisson.cdf(4,3))
0.8152632445237722
print(stats.poisson.cdf(5,3))
0.9160820579686966

0.8152632445237722
0.9160820579686966


0.9160820579686966

Calculate the following probabilities: P(X ≤5), P(X <5), P(X >4) and
P(X = 5).


In [17]:
## P(X <= 5))
print(stats.poisson.cdf(5,3))
0.9160820579686966
## P(X <5)
print(stats.poisson.cdf(4,3))
0.8152632445237722
## P(X >4)
print(1 - stats.poisson.cdf(4,3))
0.1847367554762278
## P(X = 5)
print(stats.poisson.cdf(5,3) - stats.poisson.cdf(4,3))
0.10081881344492438

0.9160820579686966
0.8152632445237722
0.1847367554762278
0.10081881344492438


0.10081881344492438

2.2 Course passing proportions

a)If a passing proportion for a course given repeatedly is assumed to be 0.80
on average, and there are 250 students who are taking the exam each time,
what is the expected value, μ and standard deviation, σ, for the number
of students who do not pass the exam for a randomly selected course?

If X is the number of students not passing a randomly selected course, this random
variable follows the binomial distribution with n = 250 and p = 0.20, so we use the
formula for the mean and variance of the binomial
μ = np = 0.2 ·250 = 50, σ2 = np(1 −p) = 250 ·0.2 ·0.8 = 40 = 6.322.
So the answer is: μ = 50 and σ = 6.32.


2.3 Notes in a box

A box contains 6 notes:
On 1 of the notes there is the number 1
On 2 of the notes there is the number 2
On 2 of the notes there is the number 3
On 1 of the notes there is the number 4
Two notes are drawn at random from the box, and the following random vari-
able is introduced: X, which describes the number of notes with the number 4
among the 2 drawn. The two notes are drawn without replacement.
a) The mean and variance for X, and P(X = 0) are?

X follows the hypergeometric distribution (2.24) with N = 6, a = 1, and n = 2 so
the mean and variance formula (2.25) for this distribution is used to find
μx = n a
N = 2/6,
and
σ2x = n a(N −a)
N2
N −n
N −1 = 2 1 ·(6 −1) ·(6 −2)
62 ·(6 −1) = 2 ·5 ·4
36 ·5 = 8/36 = 2/9.
And the hypergeometric probability formula (2-25) gives
P(X = 0) =
( 1
0
)( 5
2
)
( 6
2
) = 5 ·4 ·2
2 ·6 ·5 = 2/3.
So the correct answer is: μx = 1/3, σ2x = 2/9 and P(X = 0) = 2/3.
b) The 2 notes are now drawn with replacement. What is the probability that
none of the 2 notes has the number 1 on it?

b)The 2 notes are now drawn with replacement. What is the probability that
none of the 2 notes has the number 1 on it?

In [19]:
print(stats.binom.pmf(0,n=2,p=1/6))
0.6944444444444443

0.6944444444444443


0.6944444444444443

Another way is, since it is zero successes, then there is 5/6 probability of not getting
a success and since we want that to happen in both two draws, then we can simply
multiply the probabilities
P(X = 0) = 5
6 · 5
6 = 25
36
So the correct answer is: 0.694 or 25
36 .

2.4 Consumer survey

In a consumer survey performed by a newspaper, 20 different groceries (prod-
ucts) were purchased in a grocery store. Discrepancies between the price ap-
pearing on the sales slip and the shelf price were found in 6 of these purchased
products

a) At the same time a customer buys 3 random (different) products within
the group consisting of the 20 goods in the store. The probability that no
discrepancies occurs for this customer is?

Let X denote the number of discrepancies when purchasing 3 random (different)
products within the group of the 20 goods in the store. X then follows the hyper-
geometric distribution (NOT the binomial!!) (why not binomial: because you don’t
potentially by two goods of the same kind - you DO buy 3 DIFFERENT ones and
hence having bought one - you do NOT "put it back" again and the randomly select
- it is WITHOUT replacement). Therefore
P(X = 0) =
( 6
0
)( 14
3
)
( 20
3
) = 14 ·13 ·12 ·3 ·2
20 ·19 ·18 ·3 ·2 = 91
15 ·19 = 0.3192982.
Hence the answer is: 0.319.


2.5 Hay delivery quality

A horse owner receives 20 bales of hay in a sealed plastic packaging. To con-
trol the hay, 3 bales of hay are randomly selected, and each checked whether it
contains harmful fungal spores.
It is believed that among the 20 bales of hay 2 bales are infected with fungal
spores. A random variable X describes the number of infected bales of hay
among the three selected

a) The mean of X, (μX ), the variance of X, (σ2X ) and P(X ≥1) are

The hypergeometric distribution with N = 20, a = 2 and n = 3 is used ("sampling
without replacement"). First the mean and variance formulas for the hypergeometric
distribution gives
μx = 3 2
20 = 0.3,
and
σ2x = 3 2
20 (1 − 2
20 )(20 −3
20 −1 ) = 0.2415789.
Then we find
P(X ≥1) = 1 −P(X = 0) = 1 −
( 18
3
)( 2
0
)
( 20
3
) = 0.2842,
So the answer is: μx = 0.3, σ2x = 0.242 and P(X ≥1) = 0.2842

b) Another supplier advertises that no more than 1% of his bales of hay are
infected. The horse owner buys 10 bales of hay from this supplier, and
decides to buy hay for the rest of the season from this supplier if the 10
bales are error-free.
What is the probability that the 10 purchased bales of hay are error-free, if
1% of the bales from a supplier are infected (p1) and the probability that
the 10 purchased bales of hay are error-free, if 10% of the bales from a
supplier are infected (p10)

In [20]:
print(stats.binom.pmf(0,n=10,p=0.01))
0.9043820750088041
print(stats.binom.cdf(0,n=10,p=0.01))
0.9043820750088045

0.9043820750088041
0.9043820750088045


0.9043820750088045

In [21]:
print(stats.binom.pmf(0,n=10,p=0.1))
0.34867844009999993
print(stats.binom.cdf(0,n=10,p=0.1))
0.34867844009999993

0.3486784401
0.34867844009999993


0.34867844009999993

This 10 independent events in series where each must be non-infected, ı.e. a success
is then 1 −0.01 and 1 −0.1, and can thus also be calculated simply by multiplying
the probability of success in each event
p1 = P(X1 = 0) = 0.9910 = 0.9044,
and
p10 = P(X10 = 0) = 0.9010 = 0.3487.
So the answer becomes: P1 = 0.9044 and P10 = 0.3487.

2.6 Newspaper consumer survey

In a consumer survey performed by a newspaper, 20 different groceries (prod-
ucts) were purchased in a grocery store. Discrepancies between the price ap-
pearing on the sales slip and the shelf price were found in 6 of these purchased
products.
a) Let X denote the number of discrepancies when purchasing 3 random (dif-
ferent) products within the group of the 20 products in the store. What is
the mean and variance of X?

We must use the hypergeometric distribution, since we draw n = 3 of the N = 20
products, where a = 6 have discrepancies, with no replacement (we cannot draw
the same product twice).
The mean and variance of the hypergeometric distribution is found in Theorem 2.25,
thus
μX = n · a
N = 3 6
20 = 0.90,
and
σ2X = n · a
N
(
1 − a
N
)(N −n
N −1
)
= 3 6
20 ·
(
1 − 6
20
)(20 −3
20 −1
)
= 0.5636842.
Hence the answer is: μX = 0.90 and σ2X = 0.56.

2.7 A fully automated production

On a large fully automated production plant items are pushed to a side band
at random time points, from which they are automatically fed to a control unit.
The production plant is set up in such a way that the number of items sent to
the control unit on average is 1.6 item pr. minute. Let the random variable X
denote the number of items pushed to the side band in 1 minute. It is assumed
that X follows a Poisson distribution.

a) What is the probability that there will arrive more than 5 items at the con-
trol unit in a given minute is?

With λ = 1.6, we find that
P(X >5) = 1 −P(X ≤5) = 1 −0.994 = 0.006
where the 0.994 can be found with Python by:

In [22]:
print(1-stats.poisson.cdf(k=5,mu=1.6))
0.006040291111581331

0.006040291111581331


0.006040291111581331

So the answer is: approximately 0.6%

b) What is the probability that no more than 8 items arrive to the control unit
within a 5-minute period?

With λ5minutes = 8, we find that
P(X ≤8) = 0.593
where the 0.593 can be found by Python:

In [23]:
print(stats.poisson.cdf(k=8,mu=8))
0.5925473414375915

0.5925473414375915


0.5925473414375915

So the answer is: approximately 59.3%.

2.8 Call center staff

The staffing for answering calls in a company is based on that there will be 180
phone calls per hour randomly distributed. If there are 20 calls or more in a
period of 5 minutes the capacity is exceeded, and there will be an unwanted
waiting time, hence there is a capacity of 19 calls per 5 minutes.

What is the probability that the capacity is exceeded in a random period
of 5 minutes?

The 60 minutes mean of 180 calls corresponds to a 5 minutes mean of μ5min =
180/12 = 15 and the event of exceeding capacity is the event of observing at least
20 calls within 5 minutes. Let X represent the number of calls within a randomly
chosen 5 minutes interval, then we need to find P(X ≥20), which in Python

In [24]:
print(1-stats.poisson.cdf(k=19, mu=15))
0.12478121503252493

0.12478121503252504


0.12478121503252493

So the correct answer is: P(X ≥20) = 0.125, where X ∼Po(15).


b) If the probability should be at least 99% that all calls will be handled with-
out waiting time for a randomly selected period of 5 minutes, how large
should the capacity per 5 minutes then at least be?

Let X (as above) represent the number of calls in a randomly chosen 5 minutes in-
terval, ı.e. X ∼ Po(15). It is required that
P(“All calls will be handled”) = P(X ≤ xcapacity ) ≥0.99
where xcapacity must be the smallest capacity which keeps the probability above 0.99.
Using Python to find P(X ≤ xcapacity ) for 22,23,. . . ,26:

In [25]:
print(stats.poisson.cdf(k=[22,23,24,25,26], mu=15))
[0.967 0.981 0.989 0.994 0.997]
# Or directly using the 99th quantile
print(stats.poisson.ppf(q=0.99, mu=15))
25.0

SyntaxError: invalid syntax. Perhaps you forgot a comma? (2663190083.py, line 2)

shows that the first (smallest) capacity level achieving this is 25.
So the correct answer is: the capacity must be at least 25 per 5 minutes